shell/platform/darwin/ios/framework/Source/FlutterViewController.mm

```diff
@@ -1544,21 +1544,17 @@ - (void)performOrientationUpdate:(UIInterfaceOrientationMask)new_preferences {
           continue;
         }
         UIWindowScene* windowScene = (UIWindowScene*)scene;
-        UIInterfaceOrientationMask currentInterfaceOrientation =
-            1 << windowScene.interfaceOrientation;
-        if (!(_orientationPreferences & currentInterfaceOrientation)) {
```

Comment:
These are the lines that were removed.  Always set `-requestGeometryUpdateWithPreferences:` if the preference mask changes since the OS should handle orientation based on those preferences.

Reply:
Hmmmm, will this fix the issue?   

```
if (_orientationPreferences & currentInterfaceOrientation) // removed `!`
```

Reply:
If the current preference is "portrait" and the current orientation is "portrait", and the preference is updated to "all orientations", then we always need to set the preference so as soon as the user rotates to landscape, it works.  Likewise if the preference is "all orientations" and the current orientation is "portrait", and the preference is updated to "just portrait" then we also need to set it so that when the user tries to rotate to landscape it stays in portrait.

That's the bug this is fixing--it was only updating the preference based on the current orientation.